cmd/internal/utils/validate.go

```diff
@@ -53,7 +66,31 @@ func CheckRegistryConnection(ctx context.Context, cred *auth.Credential, regName
 		return err
 	}
 
-	sp.Success(fmt.Sprintf("Remote registry %q is reachable", regName))
+	sp.Success(fmt.Sprintf("Remote registry %q implements docker registry API V2", regName))
+	printer.Verbosef("Proceeding as user %q", cred.Username)
 
 	return nil
 }
+
+func checkRegistryUnauthenticated(regName string) error {
+	url := fmt.Sprintf("https://%s/v2/", regName)
+	client := &http.Client{
+		Timeout: 10 * time.Second,
```

Comment:
Magic number?

Reply:
Nice catch, change the client. Using a context.